### PR TITLE
Feature/issue2637

### PR DESCRIPTION
**Pull request by salomon-j**

### CODE REVIEW

**Review by chrisala — CHANGES_REQUESTED** [review]
This implementation will allow unsubmitted reports with data to be completely deleted by an end date change.

If the last report with data is not approved, the date should be allowed to be changed such that it's > the report start date.
If the last report with data is submitted/approved, the date must be >= the report end date.

**Comment by salomon-j** [comment]
I overlooked to test that scenario, mostly tested wherein the new end date is always after the report start date.
I'm currently debugging to find the method or part of code that will not delete the unsubmitted report with data  if the change end date is before the report start date.

File: grails-app/services/au/org/ala/merit/ProjectService.groovy [hunk 1]
```diff
@@ -2052,4 +2059,8 @@ class ProjectService  {
     boolean canBulkRegenerateReports(Map project) {
         return project.reports.size() > 0
     }
+
+    boolean excludesNotApproved(Map report) {
```
Comment (chrisala) [inline grails-app/services/au/org/ala/merit/ProjectService.groovy:2063]: Is this used?  It also looks like a duplicate of the method in the ReportService

File: grails-app/services/au/org/ala/merit/ProjectService.groovy [hunk 2]
```diff
@@ -774,8 +774,8 @@ class ProjectService  {
             // We allow reports to be generated up to 24 hours after the end of a project due to
             // project end dates being 00:00 of the last day of the project instead of 23:59:...
             String plannedEndDateAlignedToReportingSchedule = DateUtils.format(DateUtils.parse(plannedEndDate).plusDays(1))
-            if (lastReport && plannedEndDateAlignedToReportingSchedule < lastReport.toDate) {
-                message = "The project end date must be on or after ${DateUtils.isoToDisplayFormat(lastReport.toDate)}"
+            if (lastReport && plannedEndDateAlignedToReportingSchedule < lastReport.fromDate) {
```
Comment (chrisala) [inline grails-app/services/au/org/ala/merit/ProjectService.groovy:777]: This allows the report to be completely deleted.
It should also take into account the minimum report length setting in the project config.

**Review by chrisala — CHANGES_REQUESTED** [review]

**Pull request merged**